extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchBuildAggregator.java

```diff
@@ -83,6 +80,15 @@ public ArrayOfDoublesSketchBuildAggregator(
   @Override
   public void aggregate()
   {
+    if (values == null) {
+      values = new double[valueSelectors.length];
+    }
+
+    if (sketch == null) {
```

Comment:
@gianm : while checking something else, I saw that this change could throw an exception if `get` is called on an aggregator without ever calling `aggregate`. 
We can fix that and maintain the old behavior by adding this logic in `get` method as well.

Reply:
Ah, thanks for pointing this out. Fixed here: https://github.com/apache/druid/pull/16330